openrr-client/src/clients/ik_client.rs

```diff
@@ -235,3 +239,87 @@ pub fn check_ik_with_interpolation_and_constraints(
     }
     Ok(traj)
 }
+
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct IkSolverConfig {
+    pub root_node_name: Option<String>,
+    pub ik_target: String,
+    pub use_random_ik: bool,
+    pub allowable_position_error_m: f64,
+    pub allowable_angle_error_rad: f64,
+    pub jacobian_multiplier: f64,
+    pub num_max_try: usize,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct IkClientConfig {
+    pub name: String,
+    pub client_name: String,
+    pub ik_solver_config: IkSolverConfig,
+}
+
+pub fn create_ik_solver_with_chain(
+    full_chain: &k::Chain<f64>,
+    config: &IkSolverConfig,
+) -> IkSolverWithChain {
+    let chain = if let Some(root_node_name) = &config.root_node_name {
+        k::SerialChain::from_end_to_root(
+            full_chain.find(&config.ik_target).unwrap(),
+            full_chain.find(root_node_name).unwrap(),
+        )
+    } else {
+        k::SerialChain::from_end(full_chain.find(&config.ik_target).unwrap())
+    };
+
+    let parameters = IkSolverParameters {
+        allowable_position_error: config.allowable_position_error_m,
+        allowable_angle_error: config.allowable_angle_error_rad,
+        jacobian_multiplier: config.jacobian_multiplier,
+        num_max_try: config.num_max_try,
+    };
+
+    IkSolverWithChain::new(
+        chain,
+        if config.use_random_ik {
+            Arc::new(create_random_jacobian_ik_solver(&parameters))
+        } else {
+            Arc::new(create_jacobian_ik_solver(&parameters))
+        },
+    )
+}
+
+pub fn create_ik_client(
+    client: ArcJointTrajectoryClient,
+    full_chain: &k::Chain<f64>,
+    config: &IkSolverConfig,
+) -> IkClient<ArcJointTrajectoryClient> {
```

Comment:
Comment:
It might be better to return `Result<IkClient<>>` to remove `unwrap()` someday.
The function name will become `fn try_create_ik_client`.
Or add new constructor for `IkClient::try_new_wtih_config()`.